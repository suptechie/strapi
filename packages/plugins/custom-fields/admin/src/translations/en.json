{
  "input.format": "HEX",
  "color-picker.label": "Color Picker",
  "color-picker.description": "Select any color" 
}
