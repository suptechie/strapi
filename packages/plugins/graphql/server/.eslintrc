{
  "root": true,
  "extends": ["custom/back/typescript"],
  "parserOptions": {
    "project": ["./server/tsconfig.eslint.json"]
  },
  "rules": {
    "import/extensions:": "off"
  }
}
