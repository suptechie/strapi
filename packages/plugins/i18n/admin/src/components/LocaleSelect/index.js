import React from 'react';
import styled from 'styled-components';
import { Select, Option } from '@strapi/parts/Select';
import { Loader } from '@strapi/parts/Loader';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';
import useLocales from '../../hooks/useLocales';
import useDefaultLocales from '../../hooks/useDefaultLocales';
import { getTrad } from '../../utils';

const SmallLoader = styled(Loader)`
  img {
    height: 1rem;
    width: 1rem;
  }
`;

/**
 * The component is memoized and needs a useCallback over the onLocaleChange and
 * onClear props to prevent the Select from re-rendering N times when typing on a specific
 * key in a formik form
 */
const LocaleSelect = React.memo(({ value, onLocaleChange, error, onClear }) => {
  const { formatMessage } = useIntl();
  const { defaultLocales, isLoading } = useDefaultLocales();
  const { locales } = useLocales();

  const options = (defaultLocales || [])
    .map(locale => ({
      label: locale.name,
      value: locale.code,
    }))
    .filter(({ value: v }) => {
      const foundLocale = locales.find(({ code }) => code === v);

      return !foundLocale;
    });

  const computedValue = value || '';

  return (
    <Select
      startIcon={isLoading ? <SmallLoader>Loading the locales...</SmallLoader> : undefined}
      aria-busy={isLoading}
      label={formatMessage({
        id: getTrad('Settings.locales.modal.locales.label'),
        defaultMessage: 'Locales',
      })}
      onClear={value ? onClear : undefined}
      clearLabel={formatMessage({
        id: 'clearLabel',
        defaultMessage: 'Clear',
      })}
      error={error}
      value={computedValue}
      onChange={selectedLocaleKey => {
        const selectedLocale = options.find(locale => locale.value === selectedLocaleKey);

        if (selectedLocale) {
          onLocaleChange({ code: selectedLocale.value, displayName: selectedLocale.label });
        }
      }}
    >
      {isLoading
        ? null
        : options.map(option => (
          <Option value={option.value} key={option.value}>
            {option.label}
          </Option>
          ))}
    </Select>
  );
});

LocaleSelect.defaultProps = {
  error: undefined,
  value: undefined,
  onClear: () => undefined,
  onLocaleChange: () => undefined,
};

LocaleSelect.propTypes = {
  error: PropTypes.string,
  onClear: PropTypes.func,
  onLocaleChange: PropTypes.func,
  value: PropTypes.string,
};

export default LocaleSelect;
