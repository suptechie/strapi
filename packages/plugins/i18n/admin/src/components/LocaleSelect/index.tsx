import * as React from 'react';

import { Combobox, ComboboxOption } from '@strapi/design-system';
import { useIntl } from 'react-intl';

import useDefaultLocales from '../../hooks/useDefaultLocales';
import useLocales from '../../hooks/useLocales';
import { getTrad } from '../../utils';

type LocaleSelectProps = {
  error?: string;
  onClear?: () => void;
  onLocaleChange: (locale: { code: string; displayName: string }) => void;
  value?: string;
};

/**
 * The component is memoized and needs a useCallback over the onLocaleChange and
 * onClear props to prevent the Select from re-rendering N times when typing on a specific
 * key in a formik form
 */
const LocaleSelect = ({
  value,
  onClear = () => {},
  onLocaleChange = () => {},
  error,
}: LocaleSelectProps) => {
  const { formatMessage } = useIntl();
  const { defaultLocales, isLoading } = useDefaultLocales();
  const { locales } = useLocales();

  const options = (defaultLocales || [])
    .map((locale: any) => ({
      label: locale.name,
      value: locale.code,
    }))
    .filter(({ value: v }: any) => {
      const foundLocale = locales.find(({ code }: any) => code === v);

      return !foundLocale || foundLocale.code === value;
    });

  const computedValue = value || '';

  return (
    <Combobox
      aria-busy={isLoading}
      error={error}
      label={formatMessage({
        id: getTrad('Settings.locales.modal.locales.label'),
        defaultMessage: 'Locales',
      })}
      value={computedValue}
      onClear={value ? onClear : undefined}
      onChange={(selectedLocaleKey) => {
        const selectedLocale = options.find((locale: any) => locale.value === selectedLocaleKey);

        if (selectedLocale) {
          onLocaleChange({ code: selectedLocale.value, displayName: selectedLocale.label });
        }
      }}
      placeholder={formatMessage({
        id: 'components.placeholder.select',
        defaultMessage: 'Select',
      })}
    >
      {options.map((option: any) => (
        <ComboboxOption value={option.value} key={option.value}>
          {option.label}
        </ComboboxOption>
      ))}
    </Combobox>
  );
};

export default React.memo(LocaleSelect);
