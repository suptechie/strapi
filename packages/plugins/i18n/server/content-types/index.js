'use strict';

const localeModel = require('./locale');

module.exports = {
  [localeModel.schema.info.singularName]: localeModel,
};
