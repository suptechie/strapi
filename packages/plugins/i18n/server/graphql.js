'use strict';

const { prop, propEq, identity } = require('lodash/fp');

const LOCALE_SCALAR_TYPENAME = 'Locale';
const LOCALE_ARG_PLUGIN_NAME = 'I18NLocaleArg';

module.exports = ({ strapi }) => ({
  register() {
    const { service: getGraphQLService } = strapi.plugin('graphql');
    const { service: getI18NService } = strapi.plugin('i18n');

    const getCreateLocalizationMutationType = contentType => {
      const { getTypeName } = getGraphQLService('utils').naming;

      return `create${getTypeName(contentType)}Localization`;
    };

    getGraphQLService('extension')
      .for('content-api')
      .use(({ nexus, typeRegistry }) => {
        const i18nLocaleArgPlugin = createI18nLocaleArgPlugin({ nexus, typeRegistry });
        const i18nLocaleScalar = createLocaleScalar({ nexus });
        const createLocalizationMutations = createCreateLocalizationMutations({
          nexus,
          typeRegistry,
        });

        return {
          plugins: [i18nLocaleArgPlugin],
          types: [i18nLocaleScalar, createLocalizationMutations],
        };
      });

    const createLocaleScalar = ({ nexus }) => {
      const locales = getI18NService('iso-locales').getIsoLocales();

      return nexus.scalarType({
        name: LOCALE_SCALAR_TYPENAME,

        description: 'A string used to identify an i18n locale',

        serialize: identity,
        parseValue: identity,

        parseLiteral(ast) {
          if (ast.kind !== 'StringValue') {
            throw new TypeError('Locale cannot represent non string type');
          }

          const isValidLocale = locales.find(propEq('code', ast.value));

          if (!isValidLocale) {
            throw new TypeError('Unknown locale supplied');
          }

          return ast.value;
        },
      });
    };

    const createCreateLocalizationMutations = ({ nexus, typeRegistry }) => {
      const { KINDS } = getGraphQLService('constants');
      const { isLocalizedContentType } = getI18NService('content-types');

      const localizedContentTypes = typeRegistry
        .where(
          ({ config }) => config.kind === KINDS.type && isLocalizedContentType(config.contentType)
        )
        .map(prop('config.contentType'));

      return localizedContentTypes.map(ct => createCreateLocalizationMutation(ct, { nexus }));
    };

    const createCreateLocalizationMutation = (contentType, { nexus }) => {
      const { getEntityResponseName, getContentTypeInputName } = getGraphQLService('utils').naming;
      const { createCreateLocalizationHandler } = getI18NService('core-api');

      const responseType = getEntityResponseName(contentType);
      const mutationName = getCreateLocalizationMutationType(contentType);

      const resolverHandler = createCreateLocalizationHandler(contentType);

      return nexus.extendType({
        type: 'Mutation',

        definition(t) {
          t.field(mutationName, {
            type: responseType,

            // The locale arg will be automatically added through the i18n graphql extension
            args: {
              id: 'ID',
              data: getContentTypeInputName(contentType),
            },

            async resolve(parent, args) {
              const value = await resolverHandler(args);

              return { value, info: { args, resourceUID: contentType.uid } };
            },
          });
        },
      });
    };

    const createI18nLocaleArgPlugin = ({ nexus, typeRegistry }) => {
      const { isLocalizedContentType } = getI18NService('content-types');

      const addLocaleArg = config => {
        const { parentType } = config;

        // Only target queries or mutations
        if (parentType !== 'Query' && parentType !== 'Mutation') {
          return;
        }

        const contentType = typeRegistry.get(config.type).config.contentType;

        const createLocalizationMutationType = getCreateLocalizationMutationType(contentType);

        // Ignore non-localized content types
        if (
          !isLocalizedContentType(contentType) ||
          // Don't add the "locale" arg on "create localization" mutations
          config.name === createLocalizationMutationType
        ) {
          return;
        }

        config.args.locale = nexus.arg({ type: LOCALE_SCALAR_TYPENAME });
      };

      return nexus.plugin({
        name: LOCALE_ARG_PLUGIN_NAME,

        onAddOutputField(config) {
          // Add the locale arg to the queries on localized CTs
          addLocaleArg(config);
        },
      });
    };
  },
});
