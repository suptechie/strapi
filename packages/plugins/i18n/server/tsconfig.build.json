{
  "extends": "./server/tsconfig.json",
  "include": ["./server/src"],
  "exclude": ["node_modules", "**/*.test.ts"],
  "compilerOptions": {
    "outDir": "./dist/server"
  }
}
