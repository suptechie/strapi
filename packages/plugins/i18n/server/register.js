'use strict';

const _ = require('lodash');

const validateLocaleCreation = require('./controllers/validate-locale-creation');
const { getService } = require('./utils');

// const fieldMigration = require('./migrations/field');
const enableContentType = require('./migrations/content-type/enable');
const disableContentType = require('./migrations/content-type/disable');

module.exports = ({ strapi }) => {
  extendLocalizedContentTypes(strapi);
  addContentManagerLocaleMiddleware(strapi);
  addContentTypeSyncHooks(strapi);

  // TODO:
  // strapi.db.migrations.register(fieldMigration);
};

/**
 * Adds middleware on CM creation routes to use i18n locale passed in a specific param
 * @param {Strapi} strapi
 */
const addContentManagerLocaleMiddleware = strapi => {
  strapi.server.router.use('/content-manager/collection-types/:model', (ctx, next) => {
    if (ctx.method === 'POST') {
      return validateLocaleCreation(ctx, next);
    }

    return next();
  });

  strapi.server.router.use('/content-manager/single-types/:model', (ctx, next) => {
    if (ctx.method === 'PUT') {
      return validateLocaleCreation(ctx, next);
    }

    return next();
  });
};

/**
 * Adds hooks to migration content types locales on enable/disable of I18N
 * @param {Strapi} strapi
 */
const addContentTypeSyncHooks = strapi => {
  strapi.hook('strapi::content-types.beforeSync').register(disableContentType);
  strapi.hook('strapi::content-types.afterSync').register(enableContentType);
};

/**
 * Adds locale and localization fields to localized content types
 * @param {Strapi} strapi
 */
const extendLocalizedContentTypes = strapi => {
  const contentTypeService = getService('content-types');
  const coreApiService = getService('core-api');

  Object.values(strapi.contentTypes).forEach(contentType => {
    if (contentTypeService.isLocalizedContentType(contentType)) {
      const { attributes } = contentType;

      _.set(attributes, 'localizations', {
        writable: true,
        private: false,
        configurable: false,
        visible: false,
        type: 'relation',
        relation: 'oneToMany',
        target: contentType.uid,
      });

      _.set(attributes, 'locale', {
        writable: true,
        private: false,
        configurable: false,
        visible: false,
        type: 'string',
      });

      coreApiService.addCreateLocalizationAction(contentType);
      // coreApiService.addGraphqlLocalizationAction(contentType); // TODO: to implement
    }
  });

  if (strapi.plugin('graphql')) {
    require('./graphql')({ strapi }).register();
  }

  // TODO: to implement
  // strapi.db.migrations.register(fieldMigration);
  // strapi.db.migrations.register(enableContentTypeMigration);
  // strapi.db.migrations.register(disableContentTypeMigration);
};
