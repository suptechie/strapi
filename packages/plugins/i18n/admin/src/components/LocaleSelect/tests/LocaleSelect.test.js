import React from 'react';
import { ThemeProvider, lightTheme } from '@strapi/parts';
import { render as renderTL, fireEvent, waitFor, screen } from '@testing-library/react';
import { QueryClientProvider, QueryClient } from 'react-query';
import { Provider } from 'react-redux';
import { combineReducers, createStore } from 'redux';
import en from '../../../translations/en.json';
import server from './server';
import reducers from '../../../hooks/reducers';
import LocaleSelect from '..';

jest.mock('../../../utils', () => ({
  getTrad: x => x,
}));

jest.mock('react-intl', () => ({
  FormattedMessage: ({ id }) => id,
  useIntl: () => ({ formatMessage: jest.fn(({ id }) => en[id]) }),
}));

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useNotification: jest.fn(),
}));

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },
});

const store = createStore(combineReducers(reducers));

const render = props =>
  renderTL(
    <QueryClientProvider client={queryClient}>
      <Provider store={store}>
        <ThemeProvider theme={lightTheme}>
          <LocaleSelect {...props} />
        </ThemeProvider>
      </Provider>
    </QueryClientProvider>,
    { container: document.body }
  );

describe('LocaleSelect', () => {
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('shows an aria-busy element when loading the data', async () => {
    const { container } = render();

    expect(container.firstChild).toMatchInlineSnapshot(`
      .c5 {
        padding-left: 12px;
      }

      .c2 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: row;
        -ms-flex-direction: row;
        flex-direction: row;
        -webkit-align-items: center;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
      }

      .c6 {
        background: transparent;
        border: none;
        position: relative;
        z-index: 1;
      }

      .c6 svg {
        height: 0.6875rem;
        width: 0.6875rem;
      }

      .c6 svg path {
        fill: #666687;
      }

      .c7 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        background: none;
        border: none;
      }

      .c7 svg {
        width: 0.375rem;
      }

      .c1 {
        font-weight: 500;
        font-size: 0.75rem;
        line-height: 1.33;
        color: #32324d;
      }

      .c3 {
        position: relative;
        border: 1px solid #dcdce4;
        padding-right: 12px;
        padding-left: 12px;
        border-radius: 4px;
        background: #ffffff;
      }

      .c3:focus-within {
        outline: 2px solid #4945ff;
        outline-offset: 2px;
        box-shadow: revert;
      }

      .c4 {
        min-height: 2.5rem;
        border: none;
        -webkit-flex: 1;
        -ms-flex: 1;
        flex: 1;
        font-size: 0.875rem;
        color: #32324d;
      }

      .c4:focus-visible {
        outline: none;
        box-shadow: none;
        outline-offset: 0;
      }

      .c4[aria-disabled='true'] {
        background: inherit;
        color: inherit;
      }

      .c0 {
        display: -webkit-box;
        display: -webkit-flex;
        display: -ms-flexbox;
        display: flex;
        -webkit-flex-direction: column;
        -ms-flex-direction: column;
        flex-direction: column;
      }

      .c0 > * {
        margin-top: 0;
        margin-bottom: 0;
      }

      .c0 > * + * {
        margin-top: 4px;
      }

      <div>
        <div
          class="c0"
        >
          <label
            class="c1"
            for="combobox-1"
            id="combobox-1-label"
          >
            Locales
          </label>
          <div
            class="c2 c3"
          >
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-busy="true"
              aria-controls="combobox-1-listbox"
              aria-describedby="combobox-1-label"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="listbox"
              class="c4"
              id="combobox-1"
              placeholder="Select or enter a value"
              role="combobox"
              type="text"
              value=""
            />
            <div
              class="c2"
            >
              <button
                aria-hidden="true"
                class="c5 c6 c7"
                tabindex="-1"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 14 8"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                    fill="#32324D"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    `);
  });

  it('only shows the locales that have not already been used', async () => {
    render();

    await waitFor(() =>
      expect(screen.queryByText('Loading the available locales...')).not.toBeInTheDocument()
    );
    fireEvent.click(screen.getByLabelText('Locales'));
    await waitFor(() => screen.getByText('Afrikaans (af)'));

    expect(screen.getByText('Afrikaans (af)')).toBeVisible();
    expect(screen.getByText('French (fr)')).toBeVisible();
    expect(screen.queryByText('English (en)')).toBeFalsy();
  });

  it('brings back an object of code and displayName keys when changing', async () => {
    const onLocaleChangeSpy = jest.fn();
    render({ onLocaleChange: onLocaleChangeSpy });

    await waitFor(() =>
      expect(screen.queryByText('Loading the available locales...')).not.toBeInTheDocument()
    );
    fireEvent.click(screen.getByLabelText('Locales'));
    await waitFor(() => screen.getByText('Afrikaans (af)'));
    fireEvent.click(screen.getByText('French (fr)'));

    expect(onLocaleChangeSpy).toBeCalledWith({ code: 'fr', displayName: 'French (fr)' });
  });
});
