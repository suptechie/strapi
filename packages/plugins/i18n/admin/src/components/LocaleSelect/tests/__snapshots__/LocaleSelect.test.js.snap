// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSelect shows an aria-busy element when loading the data 1`] = `
.c2 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 12px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  position: relative;
  border: 1px solid #dcdce4;
  padding-right: 12px;
  padding-left: 12px;
  border-radius: 4px;
  background: #ffffff;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: 16px;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c4[data-disabled] {
  color: #666687;
  background: #eaeaef;
  cursor: not-allowed;
}

.c4:focus-visible {
  outline: none;
}

.c4:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c5 {
  width: 100%;
  font-size: 0.875rem;
  color: #32324d;
  min-height: 2.5rem;
  border: none;
  background-color: transparent;
}

.c5:focus-visible {
  outline: none;
}

.c5[aria-disabled='true'] {
  cursor: inherit;
}

.c7 > svg {
  width: 0.375rem;
}

.c7 > svg > path {
  fill: #666687;
}

.c7[aria-disabled='true'] {
  cursor: inherit;
}

.c8[data-selected] .c1 {
  color: #4945ff;
  font-weight: bold;
}

.c8[data-highlighted] .c1 {
  color: #4945ff;
  font-weight: bold;
}

.c9[data-selected] .c1 {
  color: #4945ff;
  font-weight: bold;
}

.c9[data-highlighted] .c1 {
  color: #4945ff;
  font-weight: bold;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <label
      class="c1 c2 c3"
      for="1"
    >
      Locales
    </label>
    <div
      class="c4"
    >
      <input
        aria-autocomplete="both"
        aria-controls="radix-0"
        aria-disabled="false"
        aria-expanded="false"
        aria-invalid="false"
        aria-labelledby="1-hint 1-error"
        aria-required="false"
        class="c5"
        data-placeholder=""
        data-state="closed"
        id="1"
        placeholder="Select"
        role="combobox"
        type="text"
        value=""
      />
      <span
        class="c6"
      >
        <button
          aria-controls="radix-0"
          aria-disabled="false"
          aria-expanded="false"
          aria-hidden="true"
          class="c7"
          tabindex="-1"
          type="button"
        >
          <svg
            fill="none"
            height="1rem"
            viewBox="0 0 14 8"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
              fill="#32324D"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </span>
    </div>
  </div>
</div>
`;
